python/pyiceberg/exceptions.py

```diff
@@ -110,3 +110,7 @@ class CommitFailedException(RESTError):
 
 class CommitStateUnknownException(RESTError):
     """Commit failed due to unknown reason."""
+
+
+class InvalidConfigurationError(Exception):
```

Comment:
Let's remove this one and just throw a `ValueError`. We also throw a ValueError when the catalog configuration is not set properly. We try to avoid introducing a lot of custom exceptions.